src/main/java/io/cdap/plugin/google/sheets/source/GoogleSheetsSourceConfig.java

```diff
@@ -661,14 +662,33 @@ private List<ColumnComplexSchemaInfo> processSubHeaders(int startIndex, int leng
     return subHeaders;
   }
 
-  private String checkTitleFormat(String title, int columnIndex) {
-    if (!COLUMN_NAME.matcher(title).matches()) {
-      String defaultColumnName = ColumnAddressConverter.getColumnName(columnIndex + 1);
-      LOG.warn(String.format("Original column name '%s' doesn't satisfy column name requirements '%s', " +
-        "the default column name '%s' will be used.", title, COLUMN_NAME.pattern(), defaultColumnName));
-      return defaultColumnName;
+  private String checkTitleFormat(String title, Map<String, Integer> seenFieldNames) {
+    final String replacementChar = "_";
+
+    StringBuilder cleanFieldNameBuilder = new StringBuilder();
+
+    // Remove any spaces at the end of the strings
+    title = title.trim();
+
+    // If it's an empty string replace it with BLANK
+    if (title.isEmpty()) {
+      cleanFieldNameBuilder.append("BLANK");
+    } else if (Character.isDigit(title.charAt(0))) {
+      // Prepend a col_ if the first character is a number
+      cleanFieldNameBuilder.append("col_");
+    }
+
+    // Replace all invalid characters with the replacement char
+    cleanFieldNameBuilder.append(NOT_VALID_PATTERN.matcher(title).replaceAll(replacementChar));
+
+    String cleanFieldName = cleanFieldNameBuilder.toString();
+    int count = seenFieldNames.getOrDefault(cleanFieldName, 0) + 1;
+    seenFieldNames.put(cleanFieldName, count);
```

Comment:
In original code in hydrator plugins `seenFieldNames` is updated on the basis of column names in `lowercase`, see comment here: https://github.com/cdapio/hydrator-plugins/blob/bf191b580582dc44d6a07a72727a898cdd431002/format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorUtils.java#L192-L200

I wonder if we should follow the same here?

Reply:
I think we can follow the same, if wrangler can create an issue, added a test case also with these changes